### DISCUSSION TITLE
Livewire is dropping the ordering of a collection somehow

### DISCUSSION THREAD

**Discussion by drdan18** [livewire/livewire#8728] [1. Help]
Hi, 

My code is below.  There is an odd behaviour I need help with. 

The public variable $fed_brackets is being set in the mount method with an orderBy method.   There are only five records and the bracket value is numbered 1 through 5.  They are in the database out of order.  

When the page first loads everything looks fine and the records in the collection are sorted by bracket value 1 through 5.  

The variable $gross is a wire:model value for an input.  As soon as that value changes updateGross() method is called.  

When I dd( $this->fed_brackets) as the first line in the updatedGross() method it shows that the collection $this->fed_brackets now has the data listed out of order.  It is as if it was pulled from the database without the orderBy.  

Why would the variable change at all when updatedGross() is called?  

Dan

//

class CalcTaxes extends Component
{

    public $fed_brackets; 
    public $prov_brackets; 

    public $tax_amounts = [];  
    public $tax_fed_total; 
    public $tax_prov_total;

    public $gross; 
    public $remaining_income; 

    function mount( ) 
    {
        $this->gross = 25234;

        $this->fed_brackets = Bracket::where('government','Federal')->orderBy('bracket')->get(); 

        $this->prov_brackets = Bracket::where('government','Provincial')->orderBy('bracket')->get();

        $this->calc_taxes();  
    }

    function updatedGross() 
    {
        $this->tax_amounts = []; 
        $this->calc_taxes(); 
    }
 
    function calc_taxes() 
    {

        foreach( $this->fed_brackets as $bracket )
        {
            $this->tax_amounts[] = ['gov' => 'Fed', 'bracket' => $bracket->bracket, 'taxable' => $this->gross, 'amount' => $bracket->percentage]; 
        }
        
        foreach( $this->prov_brackets as $bracket )
        {
            $this->tax_amounts[] = ['gov' => 'Prov', 'bracket' => $bracket->bracket, 'taxable' => $this->gross, 'amount' => 300]; 
        }
        
    }

    public function render()
    {
        return view('livewire.calc-taxes');
    }
}

//

**Comment by gitadminhub** [comment]
Are you sure your code does not include any additional lines where you change the order or gets data from database again?
What I can think of:
1. It should not make a difference but try: add 'asc' or 'desc' to your initial query orderBy.
2. If problem will persist and nothing will help: as temporary idea, move your query to render method but keep in mind, it can make your response time longer.

**Reply by drdan18** [reply]
Thanks for the reply.   There is no other code that touches $fed_brackets.  I just tried making the variable private.  Once the updatedGross() method is called the $fed_brackets is set to null.  Something in the updated hook is changing the collection.  

I can make the code work by using a sorted copy of the $fed_brackets 

$sorted_fed_brackets = $this->fed_brackets->sortBy('bracket');  

Then I used this $sorted_fed_brackets in my foreach loop and it works as expected.  

The original problem is still unexpected behaviour.
